### PR TITLE
Fix Issue #81 Market Buy

### PR DESCRIPTION
**Pull request by HashWarlock**
Add optional amount to buy function that can be defined to prevent front running the buy and listing the NFT at a higher price. Addresses #81

### CODE REVIEW

File: pallets/rmrk-market/src/lib.rs [hunk 1]
```diff
@@ -424,7 +428,7 @@ pub mod pallet {
 					}
 
 					<T as pallet::Config>::Currency::unreserve(&offer.maker, offer.amount);
-					Self::do_buy(offer.maker, collection_id, nft_id, true)?;
+					Self::do_buy(offer.maker, collection_id, nft_id, None,true)?;
```
Comment (h4x3rotab) [inline pallets/rmrk-market/src/lib.rs:431]: `cargo fmt` please

**Review by h4x3rotab — COMMENTED** [review]
Didn't understand why checking the price can prevent front running. Btw which kind of front running do you refer to?

File: pallets/rmrk-market/src/lib.rs (hunk 1 above)
Comment (HashWarlock) [reply pallets/rmrk-market/src/lib.rs:431]: > Didn't understand why checking the price can prevent front running. Btw which kind of front running do you refer to?

This was based on a video tutorial I saw on the Substrate Kitties pallet. It was mentioned that if the buy doesn't include the price, a `buyer` could buy the NFT, but the current `owner` could have updated the price before the `buy` is executed. So the `buyer` could end up buying at a higher listed price if the Storage is updated before the `do_buy` helper function reads the Storage value of the listed price. So if we pass in the price then the buyer will pay what is expected opposed to the price in Storage at the time `do_buy` is executed. Am I thinking about this in the right way or is this not the case in this scenario?

Comment (h4x3rotab) [reply pallets/rmrk-market/src/lib.rs:431]: Got it. It makes sense

Comment (HashWarlock) [reply pallets/rmrk-market/src/lib.rs]: New build will have corrected formatting errors with `cargo fmt`

**Pull request merged**